backend/src/nodes/utils/ncnn_model.py

```diff
@@ -625,3 +605,82 @@ def interpolate(self, model_b: "NcnnModel", alpha: float) -> "NcnnModel":
     @property
     def bin(self) -> bytes:
         return self.weights_bin
+
+
+class NcnnModelWrapper:
+    def __init__(self, model: NcnnModel) -> None:
+        self.model: NcnnModel = model
+        scale, in_nc, out_nc, nf = self.get_broadcast_data()
```

Comment:
I mean that it would have been possible to implement the function as:
```py
    def get_broadcast_data(self) -> Tuple[int, int, int, int, str]:
        return self.scale, self.in_nc, self.out_nc, self.nf, self.fp
```
Obviously, this is incorrect, but it would have been allowed by the type system.

Conceptually, when you call an instance method, you expect that instance to be fully initialized (= to be in a valid state). However, that was not the case. We weren't able to use *some* fields inside `get_broadcast_data` because they weren't initialized yet. This makes it pretty difficult to check whether a function is correctly implemented, because you have to manually track which instance function is allowed to access which fields.